pkg/ibmcsidriver/controller_helper.go

```diff
@@ -296,6 +335,55 @@ func getVolumeParameters(logger *zap.Logger, req *csi.CreateVolumeRequest, confi
 	return volume, nil
 }
 
+// setSecurityGroupList
+func setSecurityGroupList(volume *provider.Volume, value string) {
+	securityGroupstr := strings.TrimSpace(value)
+	securityGroupList := strings.Split(securityGroupstr, ",")
+	var securityGroups []provider.SecurityGroup
+	for _, securityGroup := range securityGroupList {
+		securityGroups = append(securityGroups, provider.SecurityGroup{ID: securityGroup})
+	}
+	volume.VPCVolume.SecurityGroups = &securityGroups
+}
+
+// setISENIEnabled
+func setISENIEnabled(volume *provider.Volume, key string, value string) error {
+	var err error
+	if value != TrueStr && value != FalseStr {
+		err = fmt.Errorf("'<%v>' is invalid, value of '%s' should be [true|false]", value, key)
+	} else {
+		if value == TrueStr {
+			volume.VPCVolume.AccessControlMode = SecurityGroup
+		} else {
+			volume.VPCVolume.AccessControlMode = VPC
+		}
+	}
+
+	return err
+}
+
+// setPrimaryIPID
+func setPrimaryIPID(volume *provider.Volume, key string, value string) error {
```

Comment:
Improvements: can we have template based implementation for this method so that we can update the volume object.

here key is not used

Reply:
We will  not do any validation here apart for such case just pass the parameter to library in next PR

Reply:
https://github.ibm.com/alchemy-containers/armada-storage/issues/4583